packages/ui/src/providers/delete-modal.provider.tsx

```diff
@@ -0,0 +1,80 @@
+import { Modal, ModalVariant, Button } from '@patternfly/react-core';
+import { FunctionComponent, PropsWithChildren, createContext, useCallback, useMemo, useRef, useState } from 'react';
+
+const DeleteModalCtx = createContext({
+  isDeleteModalOpen: false,
+});
+
+export default DeleteModalCtx;
+
+interface DeleteModalContextValue {
+  deleteConfirmation: () => Promise<boolean>;
+}
+
+export const DeleteModalContext = createContext<DeleteModalContextValue | undefined>(undefined);
+
+/**
+ * This provider is used to open the Delete Confirmation modal.
+ * The modal loads when the user clicks on the delete Routes/Kamelets of remove any Step from the Context Menu.
+ */
+export const DeleteModalContextProvider: FunctionComponent<PropsWithChildren> = (props) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const deleteConfirmationRef = useRef<{
+    resolve: (confirm: boolean) => void;
+    reject: (error: unknown) => unknown;
+  }>();
+
+  const handleCloseModal = useCallback(() => {
+    setIsModalOpen(false);
+  }, []);
```

Comment:
We need to resolve the promise with `false`, otherwise the promise will wait forever.

```suggestion
  const handleCloseModal = useCallback(() => {
    setIsModalOpen(false);
    deleteConfirmationRef.current?.resolve(false);
  }, []);
```